go/state/state_db.go

```diff
@@ -580,6 +595,24 @@ func (s *stateDB) Suicide(addr common.Address) bool {
 	return true
 }
 
+// SuicideNewContract marks the given contract account as suicided.
+// If called in the same transaction scope after CreateContract, works
+// same as Suicide, otherwise has no effect.
+func (s *stateDB) SuicideNewContract(addr common.Address) bool {
+	if !s.Exist(addr) {
+		return false
+	}
+
+	// If called in the same tx, behave as previously
+	if s.createdContracts[addr] {
+		return s.Suicide(addr)
+	}
+
+	// Otherwise do nothing
+
+	return true
```

Comment:
Should we return `false` when it act as a no-op? In fact it is not getting removed, so `false` might make more sense...

Reply:
I agree, the result should be `false`.